src/protocol/server.c

```diff
@@ -92,17 +90,50 @@ protocolServerError(ProtocolServer *this, int code, const String *message, const
     ASSERT(message != NULL);
     ASSERT(stack != NULL);
 
-    KeyValue *error = kvNew();
-    kvPut(error, VARSTR(PROTOCOL_ERROR_STR), VARINT(code));
-    kvPut(error, VARSTR(PROTOCOL_OUTPUT_STR), VARSTR(message));
-    kvPut(error, VARSTR(PROTOCOL_ERROR_STACK_STR), VARSTR(stack));
+    // Write the error and flush to be sure it gets sent immediately
+    PackWrite *error = pckWriteNew(this->write);
+    pckWriteU32P(error, protocolMessageTypeError);
+    pckWriteI32P(error, code);
+    pckWriteStrP(error, message);
+    pckWriteStrP(error, stack);
+    pckWriteEndP(error);
 
-    ioWriteStrLine(protocolServerIoWrite(this), jsonFromKv(error));
-    ioWriteFlush(protocolServerIoWrite(this));
+    ioWriteFlush(this->write);
 
     FUNCTION_LOG_RETURN_VOID();
 }
 
+/**********************************************************************************************************************************/
+ProtocolServerCommandGetResult
+protocolServerCommandGet(ProtocolServer *const this)
+{
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(PROTOCOL_SERVER, this);
+    FUNCTION_LOG_END();
+
+    ProtocolServerCommandGetResult result = {0};
+
+    MEM_CONTEXT_TEMP_BEGIN()
+    {
+        PackRead *const command = pckReadNew(this->read);
+        ProtocolMessageType type = (ProtocolMessageType)pckReadU32P(command);
```

Comment:
> by setting .defaultWrite and optionally .defaultValue

This is not right actually. If defaultWrite = true then nothing needs to be done on the read because the value was written. If only .defaultValue is set then you'd need to set the same defaultValue on read, unless that default value was the C default.

So I think it needs to be:
`Note that defaults are not stored in the pack so any defaults that were applied when writing (by setting .defaultValue) must be applied again when reading by setting .defaultValue if the default value is not a standard C default.`

Reply:
Fixed in commit 0f1c2eb4ee4674d7c22716ae3f5f0420cdc6592f